File: server/serverAPI/src/main/java/com/tempus/serverAPI/Chat/Models/Message.java
package com.tempus.serverAPI.Chat.Models;

import com.tempus.serverAPI.Models.ChatMessage;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Message extends ChatMessage {
    private String senderName;
    private String receiverGroup;
    private String receiverGroupAdmin;
    private String date;

}

File: lib/Groups/Message.dart
class Message{
  final String senderName;
  final String receiverGroup;
  final String receiverGroupAdmin;
  final String message;
  final String date;

  const Message({
    required this.senderName,
    required this.receiverGroup,
    required this.receiverGroupAdmin,
    required this.message,
    required this.date,
});


  @override
  String toString() {
    return '$senderName: $message - $date';
  }

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      senderName: json['senderName'],
      receiverGroup: json['receiverGroup'],
      receiverGroupAdmin: json['receiverGroupAdmin'],
      message: json['message'],
      date: json['date'],
    );
  }
}